Funcky/Extensions/ListExtensions.cs

```diff
@@ -0,0 +1,10 @@
+using static Funcky.Internal.ValueMapper;
+
+namespace Funcky.Extensions;
+
+public static class ListExtensions
+{
+    [Pure]
+    public static Option<int> IndexOfOrNone<TValue>(this IList<TValue> list, TValue value)
```

Comment:
I originally wanted to suggest that we implement this for `IReadOnlyList<T>`, but it does not have `IndexOf` :/

Then I looked at `IImmutableList<T>`. It has an `IndexOf` method, but with a very weird overload that takes the start index and the count, but I think we could supply both of those.
Alternatively, we could have overloads for `ImmutableList` and `ImmutableArray` separately.

Reply:
I think that should be a PR.